src/compiler/program.ts

```diff
@@ -3460,6 +3459,183 @@ namespace ts {
         }
     }
 
+    interface SymlinkedDirectory {

```

Comment:
All the new code is just moved from ConfiguredProject to program as part of this change.. Rest is threading through things